apps/cli/src/commands/apps/create.ts

```diff
@@ -1,36 +1,69 @@
 import ora from "ora";
 import c from "ansi-colors";
+import { Args, Flags } from "@oclif/core";
+import { DownloadTemplateResult, downloadTemplate } from "giget";
+import type { TemplateProvider } from "giget";
 
 import { SunodoCommand } from "../../sunodoCommand.js";
-import { createApplication } from "../../services/sunodo.js";
-import { Args, Flags } from "@oclif/core";
 
 export default class CreateApplication extends SunodoCommand {
     static description = "Create application";
 
     static examples = ["<%= config.bin %> <%= command.id %>"];
 
-    static args = { name: Args.string() };
+    static args = {
+        name: Args.string({
+            description: "application and directory name",
+            required: true,
+        }),
+    };
 
     static flags = {
-        org: Flags.string({
-            description: "organization slug of the application",
+        template: Flags.string({
+            description: "template name to use",
+            required: true,
+            options: ["javascript"],
```

Comment:
Should this be a strict list?

What if we make it open, and fail in case of not finding the directory/template?

Reply:
As you can see from te following command output, if the template doesn't exist, it just create an empty directory.

```shell
❯ ./bin/run.js apps create my-dapp --template javascript --branch feature/javascript
✔ Application created at /home/endersonmaia/Development/sunodo/sunodo/apps/cli/my-dapp
❯ ls my-dapp
Dockerfile  entrypoint.sh  package.json  README.md  src  yarn.lock
❯ ./bin/run.js apps create my-dapp --template not-found --branch feature/javascript
✖ Error creating application: Destination /home/endersonmaia/Development/sunodo/sunodo/apps/cli/my-dapp already exists.
❯ rm -rf my-dapp
❯ ./bin/run.js apps create my-dapp --template not-found --branch feature/javascript
✔ Application created at /home/endersonmaia/Development/sunodo/sunodo/apps/cli/my-dapp
❯ ls -lha my-dapp
total 8.0K
drwxrwxr-x 2 endersonmaia endersonmaia 4.0K Apr  7 12:19 .
drwxr-xr-x 9 endersonmaia endersonmaia 4.0K Apr 13 12:22 ..

```

Reply:
The download is implemented by a third party piece of code.
I suggest we live like this, with a restricted set of command options, and they implement the more complex validation when we enable third-party templates.